File: 백준/Bronze/2231. 분해합/분해합.py
N = int(input())
result = 0

for i in range(1, N):
    arr = []
    a = i
    while True:
        if a == 0:
            break
        arr.append(a%10)
        a //= 10

    if N == i + sum(arr):
        result = i
        break

print(result)

File: 백준/Bronze/2231. 분해합/분해합.cc
#include <iostream>
#include <algorithm>
#include <string>
using namespace std;

int get_digit(int num) {
  int sum = num;

  while (num > 0) {
    sum += num % 10;
    num /= 10;
  }

  return sum;
}

int main() {
  ios::sync_with_stdio(false);
  cin.tie(NULL);

  int N;
  cin >> N;
  int digit_len = to_string(N).length();
  int s = max(1, N - 9 * digit_len);

  int result = 0;
  for (int i = s; i < N; i++) {
    if (get_digit(i) == N) {
      result = i;
      break;
    }
  }

  cout << result << '\n';

  return 0;
}